openmdao/core/system.py

```diff
@@ -792,12 +796,20 @@ def _sys_apply_linear(self, mode, do_apply, vois=(None,), gs_outputs=None):
             system has access to.
         gs_outputs : dict, optional
             Linear Gauss-Siedel can limit the outputs when calling apply.
+        rel_inputs : list or None (optional)
+            List of inputs that are relevant for linear solve in a subsystem.
+            This list only includes interior connections and states.
         """
         force_fd = self.deriv_options['type'] is not 'user'
         states = self.states
         is_relevant = self._probdata.relevance.is_relevant_system
         fwd = mode == "fwd"
 
+        if rel_inputs:
+            rel_inputs = [name_relative_to(self.pathname, var) \
```

Comment:
We could probably cache this, but it will require some rework, and systems will have to know what system needs derivatives (so maybe a dict.) Your call on whether it is worth the effort with all this being replaced. Note: this information (or something like it -- it was just a list of inputs, but I can't remember if it was tuned to the subsystem calling it) was in the code early on, but got optimized away into the `do_apply` function.